src/TypeScript/EslintBridgeClient/EslintBridgeProcess.cs

```diff
@@ -158,8 +158,18 @@ private void StartServer()
 
         private void Process_Exited(object sender, EventArgs e)
         {
-            logger.LogDebug($"[eslint-bridge] [process id: {processId}] Process exited.", processId);
-            ClearProcessData();
+            // Guard against a potential race condition i.e. this event being received
+            // after an new analysis request has already created a new Process
+            if(sender is Process terminatedProcess)
+            {
+                logger.LogDebug($"[eslint-bridge] [process id: {terminatedProcess.Id}] Process exited.");
+                terminatedProcess.Exited -= Process_Exited;
+
+                if (ReferenceEquals(terminatedProcess, Process))
+                {
+                    ClearProcessData();
```

Comment:
Don't we still have a race condition here? Between referenceEquals and ClearProcessData?

I was thinking about it more -- do we really care about calling ClearProcessData in this case? The "Start" function checks `Process.HasExited`, which should evaluate to true. So perhaps we should just log a message here and do nothing. WDYT?

Reply:
Yes, there is still a race condition, but it's less likely to occur.

Perhaps we can get away with doing nothing. The `portId` won't be reset, but then that shouldn't matter since we shouldn't be logging anything else until a new process has started.

I'll remove the call to `ClearProcessData`.